### ISSUE TITLE
html.tags[tagName] is not the same as html.tagName

### ISSUE THREAD

**Issue by anpylar** [brython-dev/brython#1792]
The current code for `browser.html` assigns the generated tagdict `klass` to the dictionary `tags` and to the `html` (in both cases with the `tagName` as the key) Javascript object which later becomes the module `browser.html`

https://github.com/brython-dev/brython/blob/f9f9abf8b8cbf86493bcc45d3cbbeb323dd22f7c/www/src/builtin_modules.js#L318-L322

This means that if the class is overriden in the dictionary `browser.html.tags`, the change is not reflected in fetching the tag directly from the module by name, i.e.

```python
import browser.html as html

class MyH1:
    pass

htm.tags['H1'] = MyH1

html.tags['H1'] == html.H1  # => False
```

Pull request #1791 contains a proposal to make them be the same by converting the attribute `html.H1` (from the example) to be a `property` which fetches/sets values from/to the dictionary `html.tags`

(Warning, untested code as rebuilding the distribution relies on Python >= `3.10` which is not yet possible on this side)

**Comment by anpylar** [comment]
I have seen the pull request merged. Thx.

An additional question is:

  - Should the user also have the possibility to delete a "tagName"? (i.e. the property would also support deletion)

**Comment by PierreQuentel** [comment]
I have no objection and have merged PR #1791, then fixed a bug in it.

You can test modifications on Brython core scripts such as __`builtin_modules.js`__ without building __`brython.js`__ : just load all the core scripts in the page, as in the gallery example __`sort_table.html`__.

Not that if `MyH1` inherits `html.H1`, and then `html.H1` is set to `MyH1`, you have a sort of circular reference. I haven't seen any side effect at the moment but it might cause trouble in some situations.

**Comment by anpylar** [comment]
Attempted ... but the module seems to delete the attribute without ever hitting the delete method of the property

```python
                html[tagName] = _b_.property.$factory(
                    function(){
                        return _b_.dict.$getitem(html.tags, tagName)
                    },
                    function(self, value){
                        return _b_.dict.$setitem(html.tags, tagName, value)
                    },
                    function(self){  // delete
                        console.log('**Deleting tag')
                        try {
                            _b_.dict.__delitem__(html.tags, tagName)
                        } catch(err) {
                            if(err.__class__ === _b_.KeyError) {
                                // not in the dict, but this is a module
                                // throw AttributeError instead
                                throw _b_.AttributeError.$factory(
                                    tagName + " is not a tag")
                            }
                            throw err  // not KeyError, throw original
                        }
                        delete html[tagName]  // all ok, delete it also from module
                    }
                )
```

Neither `delattr` nor `del` do ever get to the delete portion (the `console.log` is the visual confirmation).

With both, the attribute is effectively deleted from the module, but it does of course remain in the dictionary `browser.html.tags`

**Comment by PierreQuentel** [comment]
Commit 7c2cb89 fixes a bug that was introduced by PR #1791 : `type(element)` was no more available for DOM elements, and the home page was not working any more...

In commit 3947748 I have changed the whole implementation of `html.tags`. Making it a property whose methods `fget()` and `fset()` require lookups / updates in a Python dictionary was uselessly complicated, and slow.

The commit makes `html.tags` an instance of a new class, `$B.pyobj_as_pydict`, defined in __`py_dict.js`__. It is a wrapper around the Javascript object `html`, with the interface of a Python dictionary, where lookups, updates and deletions are reflected in the original Javascript object.

I have also added a test script for the __`browser.html`__ module.

Is this ok for you ?
